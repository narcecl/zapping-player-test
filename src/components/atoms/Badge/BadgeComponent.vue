<template>
    <div class="badge">
        <slot />
    </div>
</template>

<style lang="scss" scoped>
.badge {
    background: rgba(white, 0.2);
    padding: 4px 8px;
    border-radius: 24px;
    color: white;
    border: 1px solid rgba(white, 0.1);
    font-size: var(--font-size-xsmall);
}
</style>
