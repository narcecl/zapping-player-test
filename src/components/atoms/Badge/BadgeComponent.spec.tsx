import { describe, test, expect } from 'vitest';

import { mount } from '@vue/test-utils';
import Badge from './BadgeComponent.vue';

describe('Badge', () => {
    test('Default render (button)', () => {
        const wrapper = mount(<Badge>Hello Vitest</Badge>);
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.text()).toContain('Hello Vitest');
        expect([...wrapper.element.classList]).includes('badge');
    });
});
