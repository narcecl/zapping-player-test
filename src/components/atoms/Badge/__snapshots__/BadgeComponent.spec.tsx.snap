// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > Default render (button) 1`] = `"<div data-v-bd415e2a="" class="badge">Hello Vitest</div>"`;
